File: ProgrammingFundamentals/14.DictionariesLambdaAndLINQ-Exercises/01.Phonebook/Phonebook.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace _01.Phonebook
{
    class Phonebook
    {
        static void Main(string[] args)
        {
            var input = Console.ReadLine();

            var phonebook = new Dictionary<string, string>();

            while (input != "END")
            {
                var inputTokens = input.Split(' ').ToList();

                var command = inputTokens[0];
                var name = inputTokens[1];

                if (command == "A")
                {
                    var phoneNumber = inputTokens[2];

                    phonebook[name] = phoneNumber;
                }
                else if (command == "S")
                {
                    if (phonebook.ContainsKey(name))
                    {
                        Console.WriteLine(name + " -> " + phonebook[name]);
                    }
                    else
                    {
                        Console.WriteLine($"Contact {name} does not exist.");
                    }
                }

                input = Console.ReadLine();
            }
        }
    }
}

File: Software Technologies/Java Basics - Exercises/src/Phonebook.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;
import java.util.concurrent.ConcurrentHashMap;

public class Phonebook {
    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);
        Map<String, String> phonebook = new HashMap<>();

        String input = scanner.nextLine();

        while (true){
            if (input.equals("END")){
                break;
            }
            String[] listOfAddedNameAndNumber = input.split(" ");
            String command = listOfAddedNameAndNumber[0];
            String nameOfInput = listOfAddedNameAndNumber[1];

            switch (command){

                case "A":
                    String numberOfInput = listOfAddedNameAndNumber[2];
                    phonebook.put(nameOfInput, numberOfInput);
                    break;
                case "S":
                    if (phonebook.containsKey(nameOfInput)){
                        for (String item: phonebook.keySet()) {
                            item = nameOfInput;
                            System.out.println(item + " -> " + phonebook.get(item));
                            break;
                        }
                    }else{
                        System.out.println("Contact " + nameOfInput + " does not exist.");
                    }
                    break;
            }
            input = scanner.nextLine();
        }
    }
}
